napari/plugins/exceptions.py

```diff
@@ -1,23 +1,251 @@
+import sys
+from collections import defaultdict
+from types import TracebackType
+from typing import DefaultDict, Dict, List, Optional, Tuple, Type
+import re
+import IPython.core.ultratb
+
+
+PLUGIN_ERRORS: DefaultDict[str, List['PluginError']] = defaultdict(list)
```

Comment:
Could you add a comment about how this dictionary is structured? What is it supposed to hold exactly?

Reply:
added comment
```
# This is a mapping of plugin_name -> PluginError instances
# all PluginErrors get added to this in PluginError.__init__
```